### PR TITLE
Simplify script

### PR DESCRIPTION
**Pull request by oriceon**
Script simplification, automatic identification of the installation folder, executable, etc ..

### CODE REVIEW

**Comment by oriceon** [comment]
Hi @aik099 , what do you think about this proposal?

File: PhpStorm Protocol (Win)/run_editor.js [hunk 1]
```diff
@@ -51,89 +32,58 @@ if (match) {
...
-    // read version name and product name from product-info.json
-    var versionFile = fso.OpenTextFile(toolboxDirectory + settings.folder_name + "\\product-info.json", 1, true);
-    var content = versionFile.ReadAll();
+    var locator     = new ActiveXObject('WbemScripting.SWbemLocator'),
+        service     = locator.ConnectServer('.', '/root/CIMV2'),
+        processes   = service.ExecQuery('Select * from Win32_Process WHERE Name = "' + process_name + '"');
+
+        // focus on editor based process id
+        shell.AppActivate((new Enumerator(processes)).item(0).ProcessId);
+}
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js:85]: Does this work on both 32bit and 64bit systems?

File: PhpStorm Protocol (Win)/run_editor.js [hunk 2]
```diff
@@ -51,89 +32,58 @@ if (match) {
...
 
-    eval('var productVersion = ' + content + ';');
-    settings.window_title = 'PhpStorm ' + productVersion.version;
-    editor = '"' + toolboxDirectory + settings.folder_name + '\\' + productVersion.launch[ 0 ].launcherPath.replace(/\//g, '\\') + '"';
+function debug(text) {
+    return (new ActiveXObject('WScript.Shell')).Popup(text, 0, 'Debug', 48);
 }
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js:89]: Is debug code really needed?

File: PhpStorm Protocol (Win)/run_editor.js [hunk 3]
```diff
@@ -51,89 +32,58 @@ if (match) {
...
 
-    var maxMajor = 0,
-        maxMinor = 0,
-        maxPatch = 0,
-        maxVersionFolder = "";
-    // Traverse through the fileCollection using the FOR loop
-    // read the maximum version from toolbox filesystem
-    for (var objEnum = new Enumerator(fileCollection); !objEnum.atEnd(); objEnum.moveNext()) {
-        var folderObject = ( objEnum.item() );
+                        var editor = folderObject.Path + '\\bin\\' + process_name;
```
Comment (aik099) [inline PhpStorm Protocol (Win)/run_editor.js:58]: What was the purpose of replacing the whole EXE file detection code?

**Review by aik099 — CHANGES_REQUESTED** [review]

File: PhpStorm Protocol (Win)/run_editor.js (hunk 2 above)
Comment (oriceon) [reply PhpStorm Protocol (Win)/run_editor.js:89]: No, but I prefer to keep it for as does. For debug proposes..

File: PhpStorm Protocol (Win)/run_editor.js (hunk 3 above)
Comment (oriceon) [reply PhpStorm Protocol (Win)/run_editor.js:58]: Why keep something like this if I implemented automatic detection of the folder, the executable?

Comment (aik099) [reply PhpStorm Protocol (Win)/run_editor.js:58]: Does it support all the use cases, that worked previously:

1. x86
2. x64
3. installation into the "Program Files..." folder (multiple version can be installed in parallel)
4. installation into JetBrains Toolbox folder (multiple version can be installed in parallel)

?

**Comment by aik099** [comment]
I've noticed PR close. It is really hard to keep tracking of things, when you submit PR -> I review -> you close it -> you submit another PR -> I review -> you close it :(

**Comment by oriceon** [comment]
I closed the pul request because is not qualify to your inițial implementation.

**Comment by aik099** [comment]
It's your choice. I respect it.

**Pull request closed without merging**